src/github.com/matrix-org/dendrite/clientapi/routing/routing.go

```diff
@@ -49,11 +49,11 @@ func Setup(servMux *http.ServeMux, httpClient *http.Client, cfg config.ClientAPI
 }
 
 // SetupSyncServerListeners configures the given mux with sync-server listeners
-func SetupSyncServerListeners(servMux *http.ServeMux, httpClient *http.Client, cfg config.Sync) {
+func SetupSyncServerListeners(servMux *http.ServeMux, httpClient *http.Client, cfg config.Sync, srp sync.RequestPool) {
```

Comment:
I'm a bit surprised by the types here. Shouldn't `srp` be a pointer argument? I realise the current impl just has a single pointer, so can be copied and still end up representing the same "pool", but it seems wrong to rely on that?

(likewise for other `RequestPool` params)